crabs/detection_tracking/datamodule.py

```diff
@@ -0,0 +1,241 @@
+    corresponding position in the samples.
+
+    Parameters
+    ----------
+    batch : list
+        A list of samples, where each sample can be any data structure.
+
+    Returns
+    -------
+    collated : Optional[Tuple[List[Any]]]
+        A tuple of lists, where each list contains the elements from the corresponding
+        position in the samples. If the input batch is empty or contains only `None`
+        values, the function returns `None`.
+    """
+    batch = [sample for sample in batch if sample is not None]
+
+    if len(batch) == 0:
+        return None
+
+    return tuple(zip(*batch))
+
+
+def get_train_transform() -> transforms.Compose:
+    """
+    Get a composed transformation for training data for data augmentation and
+    transform the data to tensor.
+
+    Returns
+    -------
+    transforms.Compose
+        A composed transformation that includes the specified operations.
+
+    Notes
+    -----
+    This function returns a composed transformation for use with training data.
+    The following transforms are applied in sequence:
+    1. Apply color jittering with brightness adjustment (0.5) and hue (0.3).
+    2. Apply Gaussian blur with kernel size of (5, 9) and sigma (0.1, 5.0)
+    3. Convert the image to a PyTorch tensor.
+
+    Examples
+    --------
+    >>> train_transform = get_train_transform()
+    >>> dataset = MyDataset(transform=train_transform)
+    """
+    # TODO: testing with different transforms
+    custom_transforms = []
+    custom_transforms.append(transforms.ColorJitter(brightness=0.5, hue=0.3))
+    custom_transforms.append(
+        transforms.GaussianBlur(kernel_size=(5, 9), sigma=(0.1, 5.0))
+    )
+    custom_transforms.append(transforms.ToTensor())
+
+    return transforms.Compose(custom_transforms)
+
+
+def get_eval_transform() -> transforms.Compose:
+    """
+    Get a composed transformation for evaluation data which
+    transform the data to tensor.
+
+    Returns
+    -------
+    transforms.Compose
+        A composed transformation that includes the specified operations.
+    """
+    custom_transforms = []
+    custom_transforms.append(transforms.ToTensor())
+
+    return transforms.Compose(custom_transforms)
+
+
+class myDataModule(pl.LightningDataModule):
+    """
+    Data module for handling data loading and preprocessing for object detection models.
+
+    Parameters:
+    -----------
+    main_dir : str
+        The main directory path containing the dataset.
+    annotation : str
+        The filename of the COCO annotation JSON file.
+    batch_size : int
+        Batch size for data loading.
+    seed_n : int, optional (default=42)
+        Random seed for data splitting.
+
+    Attributes:
+    -----------
+    main_dir : str
+        The main directory path containing the dataset.
+    annotation : str
+        The filename of the COCO annotation JSON file.
+    batch_size : int
+        Batch size for data loading.
+    seed_n : int
```

Comment:
The reason for having the seed here is to make sure we not using the same frames in training and evaluation. The current code has separate training and evaluation. So we run training, save the model. Then when we want to see the result, we run the evaluation, and load the model. so by setting the seed, we can control the split properly. I will in my plan to merge them into one code. so when we run the training, it will save the model and do the evaluation as well. But during development, it is always better to have them separately in my opinion.
The seed_n is a part of args now.